sentry_sdk/tracing.py

```diff
@@ -1285,10 +1283,19 @@ def __enter__(self):
             # set as the implicit current context
             self._ctx_token = context.attach(ctx)
 
+        self._scope_manager = sentry_sdk.new_scope()
+        scope = self._scope_manager.__enter__()
+        scope.span = self
+
         return self
 
     def __exit__(self, ty, value, tb):
         # type: (Optional[Any], Optional[Any], Optional[Any]) -> None
+        self._scope_manager.__exit__(ty, value, tb)
+        del (
+            self._scope_manager
+        )  # Eliminate circular reference which could prevent garbage collection
+
```

Comment:
I added this to be able to access the current span. Is this code needed? Should we consider it in a separate PR, or is it okay to include here?

Reply:
no I will take care of all scope related stuff, please remove both this and the above `new_scope`

Reply:
Done!